instance-id-handler/instanceIDhandler.go

```diff
@@ -0,0 +1,161 @@
+package instanceIDhandler
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/kubescape/k8s-interface/k8sinterface"
+	"github.com/kubescape/k8s-interface/workloadinterface"
+)
+
+type InstanceID struct {
+	apiVersion    string
+	namespace     string
+	kind          string
+	name          string
+	containerName string
+}
+
+const (
+	labelPrefix                 = "kubescape.io"
+	StringFormat                = "apiVersion-%s/namespace-%s/kind-%s/name-%s/containerName-%s"
+	LabelFormatKeyApiGroup      = labelPrefix + "/workload-api-group"
+	LabelFormatKeyApiVersion    = labelPrefix + "/workload-api-version"
+	LabelFormatKeyNamespace     = labelPrefix + "/workload-namespace"
+	LabelFormatKeyKind          = labelPrefix + "/workload-kind"
+	LabelFormatKeyName          = labelPrefix + "/workload-name"
+	LabelFormatKeyContainerName = labelPrefix + "/workload-container-name"
+)
+
+func GenerateInstanceID(w *workloadinterface.Workload) ([]InstanceID, error) {
+	instanceIDs := make([]InstanceID, 0)
+	parentKind, parentName := "", ""
+
+	if w.GetKind() != "Pod" {
+		return nil, fmt.Errorf("CreateInstanceID: workload kind must be Pod for create instance ID")
+	}
+	ownerReferences, err := w.GetOwnerReferences()
+	if err != nil {
+		return nil, err
+	}
+	if len(ownerReferences) == 0 {
+		parentKind = w.GetKind()
+		parentName = w.GetName()
+	} else {
```

Comment:
```suggestion
	} else if  ownerReferences[0].Kind == "Node" {
```
Then, you add the `else`